cunumeric/eager.py

```diff
@@ -1468,10 +1471,6 @@ def unary_reduction(
             return
         if op in _UNARY_RED_OPS:
             fn = _UNARY_RED_OPS[op]
-            if initial is None:
-                # NumPy starts using this predefined constant, instead of None,
-                # to mean no value was given by the caller
-                initial = np._NoValue  # type: ignore
```

Comment:
`initial` is never subsequently used on this branch, maybe some leftover cruft?

Reply:
This is a bug; this blocks should remain, and the `initial` value should be passed to the `fn` call right after. Otherwise the `initial` keyword doesn't work correctly in eager mode:

```
from cunumeric._ufunc import add
print(add.reduce(cn.ones((4,)), initial=10))
print(np.add.reduce(np.ones((4,)), initial=10))
```

when run w/o `-cunumeric:test` this prints:

```
4.0
14.0
```

Note that we can't use `cn.add.reduce`, because the coverage wrapper hides all methods on ufuncs besides `__call__()`, but that a question for another PR.

Reply:
Thanks mypy, sorry for ignoring you!

Reply:
That's weird, I was suspicious so I did run all the tests after this change and they passed for all modes :shrug: will put it back, but will use a conditional to avoid using np private values.

Reply:
please check https://github.com/nv-legate/cunumeric/pull/716/commits/a8151081c7fcd5abf6b1d8e8a511d109457a05ca

Reply:
You're right, it looks like `reduce(initial=, ...)` is not tested at all, possibly because `cn.add.reduce(...)` and friends are not currently accessible. We should add that test when we fix the ufunc wrapping.